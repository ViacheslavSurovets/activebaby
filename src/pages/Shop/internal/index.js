export { Categories } from './Categories';
export { CategoryItem } from './CategoryItem';
export { CategoryPage } from './CategoryPage';

