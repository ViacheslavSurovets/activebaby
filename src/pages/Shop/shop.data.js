import * as cnl from '@assets/newPrams/Chantal';
import * as crno from '@assets/newPrams/Cristiano';
import * as nxt from '@assets/newPrams/Neonex';
import * as rgo from '@assets/newPrams/Reggio';
import * as sra from '@assets/newPrams/Sierra';
import * as cpr from '@assets/newStrollers/capri';
import * as lvy from '@assets/newStrollers/levi';
import * as sfr from '@assets/newStrollers/safari';
import * as crl from '@assets/carSeats/carlo';
import * as kt from '@assets/carSeats/Kite';
import * as adp from '@assets/accessories/adapters';
import * as cmr from '@assets/accessories/cameras';
import * as trs from '@assets/accessories/tires';
import * as mcs from '@assets/accessories/mechanism';
import * as lns from '@assets/linens';

export const SHOP_DATA = [
  {
    title: 'new prams',
    routeName: 'new-prams',
    items: [
      {
        chantal: {
          title: 'chantal',
          info: 'good pram',
          images: [
            cnl.chantal1,
            cnl.chantal2,
            cnl.chantal3,
            cnl.chantal4,
            cnl.chantal5,
            cnl.chantal6,
            cnl.chantal8,
            cnl.chantal9,
            cnl.chantal10,
            cnl.chantal11,
            cnl.chantal12,
            cnl.chantal13,
            cnl.chantal14,
            cnl.chantal15,
            cnl.chantal7
          ],
          price: '1150',
          priceEcco: '1200',
          priceSpEdition: '1400'
        },
        cristiano: {
          title: 'cristiano',
          info: 'good pram',
          images: [
            crno.cristiano1,
            crno.cristiano2,
            crno.cristiano3,
            crno.cristiano4,
            crno.cristiano5,
            crno.cristiano6,
            crno.cristiano7,
            crno.cristiano8,
            crno.cristiano9,
            crno.cristiano10,
            crno.cristiano11,
            crno.cristiano12,
            crno.cristiano13,
            crno.cristiano14,
            crno.cristiano15,
            crno.cristiano16
          ],
          price: '1100',
          priceEcco: '1150',
          priceSpEdition: '1300'
        },
        neonex: {
          title: 'neonex',
          info: 'good pram',
          images: [
            nxt.neonex1,
            nxt.neonex2,
            nxt.neonex3,
            nxt.neonex4,
            nxt.neonex5,
            nxt.neonex6,
            nxt.neonex7,
            nxt.neonex8,
            nxt.neonex9,
            nxt.neonex10,
            nxt.neonex11,
            nxt.neonex12,
            nxt.neonex13,
            nxt.neonex14,
            nxt.neonex15,
            nxt.neonex16,
            nxt.neonex17,
            nxt.neonex18,
            nxt.neonex19
          ],
          price: '790',
          priceEcco: '900',
          priceSpEdition: '1100'
        },
        reggio: {
          title: 'reggio',
          info: 'goodPram',
          images: [
            rgo.reggio1,
            rgo.reggio2,
            rgo.reggio3,
            rgo.reggio4,
            rgo.reggio5,
            rgo.reggio6,
            rgo.reggio7,
            rgo.reggio8,
            rgo.reggio9,
            rgo.reggio10,
            rgo.reggio11,
            rgo.reggio12,
            rgo.reggio13,
            rgo.reggio14,
            rgo.reggio15,
            rgo.reggio16,
            rgo.reggio17,
            rgo.reggio18,
            rgo.reggio19,
            rgo.reggio20,
            rgo.reggio21,
            rgo.reggio22,
            rgo.reggio23,
            rgo.reggio24,
            rgo.reggio125,
            rgo.reggio126,
            rgo.reggio127,
            rgo.reggio28
          ],
          price: '1000',
          priceEcco: '1100',
          priceSpEdition: '1200'
        },
        sierra: {
          title: 'sierra',
          info: 'good pram',
          images: [
            sra.sierra1,
            sra.sierra2,
            sra.sierra3,
            sra.sierra4,
            sra.sierra5,
            sra.sierra6,
            sra.sierra7,
            sra.sierra8,
            sra.sierra9
          ],
          price: '1200',
          priceEcco: '1300',
          priceSpEdition: '1400'
        }
      }
    ]
  },
  {
    title: 'new strollers',
    routeName: 'new-strollers',
    items: [
      {
        title: 'Capri',
        info: 'good pram',
        images: [
          cpr.capri1,
          cpr.capri2,
          cpr.capri3,
          cpr.capri4,
          cpr.capri5,
          cpr.capri6,
          cpr.capri7,
          cpr.capri8
        ],
        price: '450',
        priceEcco: '-',
        priceSpEdition: '480'
      },
      {
        title: 'Levi',
        info: 'good pram',
        images: [
          lvy.levi1,
          lvy.levi2,
          lvy.levi3,
          lvy.levi4,
          lvy.levi5,
          lvy.levi6
        ],
        price: '360',
        priceEcco: '-',
        priceSpEdition: '-'
      },
      {
        title: 'Safary',
        info: 'good pram',
        images: [
          sfr.safari1,
          sfr.safari2,
          sfr.safari3,
          sfr.safari4,
          sfr.safari5,
          sfr.safari6,
          sfr.safari7,
          sfr.safari8,
          sfr.safari9,
          sfr.safari10,
          sfr.safari11,
          sfr.safari12,
          sfr.safari13
        ],
        price: '300',
        priceEcco: '-',
        priceSpEdition: '-'
      }
    ]
  },
  {
    title: 'car seats',
    routeName: 'car-seats',
    items: [
      {
        title: 'Carlo',
        info: 'good car seat',
        images: [
          kt.kite1,
          kt.kite2,
          kt.kite3,
          kt.kite4,
          kt.kite5,
          kt.kite6,
          kt.kite7,
          kt.kite8,
          kt.kite9,
          kt.kite10,
          kt.kite11
        ],
        price: '300',
        priceEcco: '350',
      },
      {
        title: 'Carlo',
        info: 'good car seat',
        images: [
          crl.carlo1,
          crl.carlo2,
          crl.carlo3,
          crl.carlo4,
          crl.carlo5,
          crl.carlo6
        ],
        price: '300',
        priceEcco: '350',
      }
    ]
  },
  {
    title: 'accessories',
    routeName: 'accessories',
    items: [
      {
        title: 'adapters',
        info: 'adapters kite',
        images: [adp.adapterKite],
        price: '40'
      },
      {
        title: 'adapter',
        info: 'adapter galactic',
        images: [adp.adapterGalactic],
        price: '25'
      },
      {
        title: 'adapter',
        info: 'adapter carlo',
        images: [adp.adapterCarlo],
        price: '30'
      },
      {
        title: 'adapter',
        info: 'adapter gussio',
        images: [adp.adapterGussio],
        price: '30'
      },
      {
        title: 'cameras',
        info: 'cameras',
        images: [cmr.camera],
        price: '10'
      },
      {
        title: 'tires',
        info: 'tires',
        images: [trs.tire],
        price: '15'
      },
      {
        title: 'mechanisms',
        info: 'mechanism',
        images: [mcs.mechanism],
        price: '45'
      }
    ]
  },
  {
    title: 'linens',
    routeName: 'linens',
    items: [
      {
        title: 'linens',
        info: 'good linens',
        images: [
          lns.linens1,
          lns.linens2,
          lns.linens3,
          lns.linens4,
          lns.linens5
        ],
        price: '105'
      }
    ]
  }
];
