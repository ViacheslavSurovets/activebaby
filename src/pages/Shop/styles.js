import styled from 'styled-components';

export const ShopContainer = styled.section`
  min-height: 80vh;
`;
