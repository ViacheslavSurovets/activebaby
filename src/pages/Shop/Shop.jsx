import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';
import { Categories } from './internal';
import { CategoryPage } from './internal';
import { ShopContainer } from './styles';
import { selectCollections, selectCollectionsLoaded } from '@redux/shop/shop.selecors';
import { createStructuredSelector } from 'reselect';
import { fetchCollectionsStart } from '@redux/shop/shop.actions';
import PropTypes from 'prop-types';


const Shop = ( { match, fetchCollectionsStart, collectionsLoaded } ) => {

  useEffect ( () => {
    //----------------- pasting data from firebase to redux
    fetchCollectionsStart ();
  }, [] );

  return (
    <ShopContainer>
      {
        collectionsLoaded && (
          <>
            <Route exact path={ `${ match.path }` } component={ Categories } />
            <Route path={ `${ match.path }/:collectionId` } component={ CategoryPage } />
          </>
        )
      }
    </ShopContainer>
  );
};

const mapStateToProps = createStructuredSelector ( {
  collections: selectCollections,
  collectionsLoaded: selectCollectionsLoaded
} );

const mapDispatchToProps = dispatch => ({
  fetchCollectionsStart: () => dispatch ( fetchCollectionsStart () )
});


export default connect ( mapStateToProps, mapDispatchToProps ) ( Shop );

Shop.propTypes = {
  match: PropTypes.object,
  fetchCollectionsStart: PropTypes.func,
  collectionsLoaded: PropTypes.bool
};
