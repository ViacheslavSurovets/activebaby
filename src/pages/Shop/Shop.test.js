import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Shop from './Shop';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('Shop', ()=> {
  it('Shop Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Shop  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
