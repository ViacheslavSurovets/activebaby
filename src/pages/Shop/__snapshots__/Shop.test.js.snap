// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Shop Snapshot 1`] = `ShallowWrapper {}`;
